src/components/button-toggle/button-toggle.component.tsx

```diff
@@ -25,12 +30,20 @@ export interface ButtonToggleProps
   "data-element"?: string;
   /** Identifier used for testing purposes, applied to the root element of the component. */
   "data-role"?: string;
-  /** Set the default value of the Group if component is meant to be used as uncontrolled. */
-  defaultChecked?: boolean;
   /** Remove spacing from between buttons. */
   grouped?: boolean;
-  /** Callback triggered by click event on the input. */
-  onClick?: (ev: React.MouseEvent<HTMLInputElement>) => void;
+  /** An optional string by which to identify the button in an onChange handler on the parent ButtonToggleGroup. */
+  name?: string;
+  /** Callback triggered by blur event on the button. */
+  onBlur?: (ev: React.FocusEvent<HTMLButtonElement>) => void;
+  /** Callback triggered by focus event on the button. */
+  onFocus?: (ev: React.FocusEvent<HTMLButtonElement>) => void;
+  /** Callback triggered by click event on the button. */
+  onClick?: (ev: React.MouseEvent<HTMLButtonElement>) => void;
+  /** Set the pressed state of the toggle button */
+  pressed?: boolean;
+  /** An optional string by which to identify the button in either an onClick handler, or an onChange handler on the parent ButtonToggleGroup. */
+  value?: string;
```

Comment:
IMO `value` should be required as there's no point of using this component without it. Or maybe I'm missing something :)

Reply:
that's a fair point, since the component can no longer be used uncontrolled. I'd be wary of adding yet another breaking change but as it's unlikely to break anything that wouldn't already be broken by this PR, I'm happy to do it.

Reply:
So I've realised that I can't force `value` to be a `string` (ie never `null` or `undefined`) because it's possible that no value might be selected - certainly that's possible on initial load, and if Harpal approves your suggestion of having a button be pressable to deselect then it can be true later too. We could allow `null` for that case rather than `undefined` but this requires quite a few changes and I'm not clear that there's any benefit over just keeping `undefined` being possible and having that mean "there is currently nothig selected".